sdk/extensions/chronopost/karrio/providers/chronopost/tracking.py

```diff
@@ -0,0 +1,84 @@
+import typing
+from urllib import response
+import karrio.lib as lib
+import karrio.core.models as models
+import karrio.providers.chronopost.utils as provider_utils
+import karrio.providers.chronopost.error as provider_error
+import chronopost_lib.trackingservice as chronopost
+
+
+def parse_tracking_response(
+    responses: typing.List[lib.Element],
+    settings: provider_utils.Settings,
+) -> typing.Tuple[typing.List[models.TrackingDetails], typing.List[models.Message]]:
+    tracking_nodes: typing.List[chronopost.eventInfoComp] = [
+        lib.find_element("events", response, chronopost.eventInfoComp)
+        for response in responses
+    ]
+    errors = [
+        provider_error.parse_error_response(result, settings) for result in responses
+    ]
+    tracking_details = [
+        _extract_details(tracking_nodes[t], settings, responses[t])
+        for t in range(len(tracking_nodes))
+    ]
+
+    return tracking_details, errors
+
+
+def _extract_details(
+    detail: typing.List[chronopost.eventInfoComp],
+    settings: provider_utils.Settings,
+    response: typing.List[lib.Element],
+) -> models.TrackingDetails:
```

Comment:
With the fixes above, all you will need is the node and settings as parameters
```suggestion
def _extract_details(
    detail: chronopost.eventInfoComp,
    settings: provider_utils.Settings,
) -> models.TrackingDetails:
```